include/seqan3/alignment/data_structures/aligned_sequence.hpp

```diff
@@ -0,0 +1,187 @@
+};
+//!\endCond
+
+//!\brief An exception thrown in seqan3::erase_gap when a position does not contain a gap.
+class align_erase_gap_exception : public std::runtime_error
+{
+public:
+    /*!\brief The constructor.
+     * \param[in] s The error message.
+     */
+    align_erase_gap_exception(std::string const & s) : std::runtime_error(s) {}
+};
+
+/*!\brief Generic function to insert a seqan3::gap::GAP into a sequence container.
+ * \tparam seq_type The sequence container type. that must satisfy the
+                    seqan3::sequence_container_concept. The underlying `value_type`
+                    must be assignable from seqan3::gap::GAP.
+ * \param[in,out] seq    The sequence container to modify.
+ * \param[in]     pos_it The iterator pointing to the position of where to insert a gap.
+ *
+ * \details
+ *
+ * This function delegates to the member function `insert(iterator, value)` of the
+ * container.
+ */
+template <sequence_container_concept seq_type>
+//!\cond
+    requires assignable_concept<typename seq_type::value_type, gap>
+//!\endCond
+void insert_gap(seq_type & seq, typename seq_type::const_iterator pos_it)
+{
+    seq.insert(pos_it, gap::GAP);
+}
+
+/*!\brief Generic function to insert seqan3::gap::GAP 's into a sequence container.
+ * \tparam seq_type The sequence container type. that must satisfy the
+                    seqan3::sequence_container_concept. The underlying `value_type`
+                    must be assignable from seqan3::gap::GAP.
+ * \param[in,out] seq    The sequence container to modify.
+ * \param[in]     pos_it The iterator pointing to the position of where to insert gaps.
+ * \param[in]     count  The number of gaps to insert.
+ *
+ * \details
+ *
+ * This function delegates to the member function `insert(iterator, count, value)` of the
+ * container.
+ */
+template <sequence_container_concept seq_type>
+//!\cond
+    requires assignable_concept<typename seq_type::value_type, gap>
+//!\endCond
+void insert_gap(seq_type & seq, typename seq_type::const_iterator pos_it, typename seq_type::size_type count)
+{
+    seq.insert(pos_it, count, gap::GAP);
+}
+
+
+/*!\brief Generic function to erase a gap from a sequence container.
+ * \tparam seq_type The sequence container type. that must satisfy the
+                    seqan3::sequence_container_concept. The underlying `value_type`
+                    must be assignable from seqan3::gap::GAP.
+ * \param[in,out] seq    The sequence container to modify.
+ * \param[in]     pos_it The iterator pointing to the position of where to erase a gap.
+ *
+ * \throws seqan3::align_erase_gap_exception
+ *
+ * \details
+ *
+ * This function delegates to the member function `erase(iterator)` of the
+ * container. Before delegating, the function checks if the position pointed to
+ * by \p pos_it is an actual gap and throws an exception if not.
+ */
+template <sequence_container_concept seq_type>
+//!\cond
+    requires assignable_concept<typename seq_type::value_type, gap>
```

Comment:
Yes because of the comparison `*pos_it != gap::GAP` which implicitly converts the right hand side value to the type of the left hand side by an assignment.